trieste/models/gpflow/models.py

```diff
@@ -1362,3 +1368,259 @@ def covariance_between_points(
             query_points_2=query_points_2,
             whiten=whiten,
         )
+
+
+class AR1(TrainableProbabilisticModel):
+    r"""
+    A :class:`TrainableProbabilisticModel` implementation of the AR1 model
+    from :cite:`Kennedy2000`. This is a multi-fidelity model that works with an
+    arbitrary number of fidelities. It relies on there being a linear relationship
+    between fidelities, and may not perform well for more complex relationships.
+    Precisely, it models the relationship between sequential fidelities as
+    .. math:: f_{i}(x) = \rho f_{i-1}(x) + \delta(x)
+    where :math:`\rho` is a scalar and :math:`\delta` models the residual between the fidelities.
+    All the base models used in this implementation are :class:`~gpflow.models.GPR` models.
+    Note: Currently only supports single output problems.
+    """
+
+    def __init__(
+        self,
+        fidelity_models: Sequence[GaussianProcessRegression],
+    ):
+        """
+        :param fidelity_models: List of
+            :class:`~trieste.models.gpflow.models.GaussianProcessRegression`
+            models, one for each fidelity.
+        """
+        self.num_fidelities = len(fidelity_models)
+
+        self.lowest_fidelity_signal_model = fidelity_models[0]
+        # Note: The 0th index in the below is not a residual model, and should not be used.
+        self.fidelity_residual_models: Sequence[GaussianProcessRegression] = fidelity_models
+        # set this as a Parameter so that we can optimize it
+        rho = [
+            gpflow.Parameter(1.0, trainable=True, name=f"rho_{i}")
+            for i in range(self.num_fidelities - 1)
+        ]
+        self.rho: list[gpflow.Parameter] = [
+            gpflow.Parameter(1.0, trainable=False, name="dummy_variable"),
+            *rho,
+        ]
+
+    def predict(self, query_points: TensorType) -> tuple[TensorType, TensorType]:
+        """
+        Predict the marginal mean and variance at query_points.
+
+        :param query_points: Query points with shape [N, D+1], where the
+            final column of the final dimension contains the fidelity of the query point
+        :return: mean: The mean at query_points with shape [N, P],
+                 and var: Th variance at query_points with shape [N, P]
+        """
+        (
+            query_points_wo_fidelity,
+            query_points_fidelity_col,
+        ) = check_and_extract_fidelity_query_points(
+            query_points, max_fidelity=self.num_fidelities - 1
+        )
+
+        signal_mean, signal_var = self.lowest_fidelity_signal_model.predict(
```

Comment:
just to check my understanding, this fetches the fidelity-1 level?

Reply:
No, this fetches the fidelity==0 level